src/js/components/Calendar/Calendar.js

```diff
@@ -68,262 +52,293 @@ const buildState = props => {
-              year: 'numeric',
-            })}
-            icon={<NextIcon size={size !== 'small' ? size : undefined} />}
-            disabled={!betweenDates(nextMonth, bounds)}
-            onClick={() => this.setReference(nextMonth)}
-          />
-        </Box>
-      </Box>
+          }
+        } else {
+          // have dates
+          const priorDates = dates[0].map(d => new Date(d));
+          const previousDate = new Date(lastSelectedDate);
+          const selDate = new Date(selectedDate);
+          if (selDate.getTime() === priorDates[0].getTime()) {
+            [[, nextDate]] = dates;
+            nextDates = undefined;
+          } else if (selDate.getTime() === priorDates[1].getTime()) {
+            [[nextDate]] = dates;
+            nextDates = undefined;
+          } else if (selDate.getTime() < previousDate.getTime()) {
+            if (selDate.getTime() < priorDates[0].getTime()) {
+              nextDates = [[selectedDate, dates[0][1]]];
+            } else if (selDate.getTime() > priorDates[0].getTime()) {
+              nextDates = [[dates[0][0], selectedDate]];
+            }
+          } else if (selDate.getTime() > previousDate.getTime()) {
+            if (selDate.getTime() > priorDates[1].getTime()) {
+              nextDates = [[dates[0][0], selectedDate]];
+            } else if (selDate.getTime() < priorDates[1].getTime()) {
+              nextDates = [[selectedDate, dates[0][1]]];
+            }
+          }
+        }
+
+        setDates(nextDates);
+        setDate(nextDate);
+        setActive(new Date(selectedDate));
+        setLastSelectedDate(selectedDate);
+        if (onSelect) onSelect(nextDates || nextDate);
+      },
+      [date, dates, lastSelectedDate, onSelect, range],
     );
-  };
-
-  renderDaysOfWeek = (locale, size, start) => {
-    let day = new Date(start);
-    const days = [];
-    while (days.length < 7) {
-      days.push(
-        <StyledDayContainer key={days.length} sizeProp={size}>
-          <StyledDay otherMonth sizeProp={size}>
-            {day.toLocaleDateString(locale, { weekday: 'narrow' })}
-          </StyledDay>
-        </StyledDayContainer>,
-      );
-      day = addDays(day, 1);
-    }
-    return <StyledWeek>{days}</StyledWeek>;
-  };
 
-  render() {
-    const {
-      bounds,
-      date: dateProp,
-      dates: datesProp,
-      disabled,
-      daysOfWeek,
-      firstDayOfWeek,
-      header,
-      locale,
-      onReference,
-      onSelect,
-      range,
-      showAdjacentDays,
-      size,
-      theme,
-      ...rest
-    } = this.props;
-    const { date, dates, focused, start, reference, end, slide } = this.state;
+    const renderCalendarHeader = () => {
+      const PreviousIcon =
+        size === 'small'
+          ? theme.calendar.icons.small.previous
+          : theme.calendar.icons.previous;
+
+      const NextIcon =
+        size === 'small'
+          ? theme.calendar.icons.small.next
+          : theme.calendar.icons.next;
+
```

Comment:
can we add a condition to if the theme exists before we are attempting `theme.calendar.icons.small.next`

Reply:
I just carried these over from before. And I think we've always got `theme.calendar.icons` in the base theme. Are you suggesting that if there is no theme at all we should be more defensive and have default icons imported in Calendar.js? I had thought we were moving away from that pattern and doing the icon import in the theme?

Reply:
agreed.